6211 Маркова Юлия Сергеевна  Лаб.2 Вар.16

marulias opened the pull request:
ФИО: Маркова Юлия Сергеевна 
Номер группы: 6211
Номер лабораторной: 2
Номер варианта: 16
Краткое описание решаемой задачи: Согласно варианту 16 лабораторной 2 мне надо было скачать изображения по ключевому слову "pig", задать ограничение по времени для скачивания. Изображения должны скачиваться пока не закончится время, либо пока не наберётся минимум (50 изображений). В конце вывести затраченное время.
Краткое описание работы кода: Программа скачивает до 50 изображений свиней по ключевому слову "pig" через Bing Image Search, создает CSV-файл аннотации с абсолютными и относительными путями к файлам, замеряет время выполнения и демонстрирует работу кастомного итератора для перебора путей к изображениям.

marulias committed:
lab1

marulias committed:
Update lab1.py

marulias committed:
Update lab1.py

marulias committed:
Update lab1.py

исправила код по вашим замечаниям

marulias committed:
lab2

marulias committed:
Delete lab2 directory

marulias committed:
lab2

microalan reviewed (commented):
1. Разделить на модули
2. Выделить создание аннотации и скачивание изображений в отдельные функции
3. Добавить обработку исключений

marulias committed:
Delete lab2 directory

marulias committed:
lab2

marulias committed:
Update functions.py

разделила на модули, добавила исключения и выделила аннотацию и скачивание файлов в отдельные функции

marulias committed:
Delete lab1 directory

microalan reviewed (commented):
1. Добавить type-hints и docstring
2. Убрать print из функций (кроме main), исключения пробрасывать дальше
3. Изменить функцию скачивания с ограничением по времени - сейчас она свою задачу не выполняет. Таймер должен быть в самой функции и она должна продолжать работать, пока таймер не выйдет или пока не наберётся минимальное количество изображений (тоже лучше задавать параметром со значением по умолчанию).

marulias committed:
Update functions.py

постаралась исправить замечания

marulias committed:
Update main.py

microalan reviewed (commented):
Добавить docstring

marulias committed:
Update functions.py

добавила докстринг

marulias committed:
Update main.py

microalan reviewed (approved):
Зачтено
